### DISCUSSION TITLE
QBKG12LM being recognised as QBKG11LM

### DISCUSSION THREAD

**Discussion by xhemp** [Koenkk/zigbee2mqtt#12330] [Devices]
Greetings!

I have a Xiaomi QBKG12LM being recognised as a QBKG12LM so just one key works, I have been trying to force the device to update the device via OTA with th QBKG12LM but not having any luck.
Have been trying with the override ota configuration(including force: true)  but haven't been having any luck with it.
Is there any particular way of making it happen?

Many thanks!

**Comment by Koenkk** [comment]
Can you provide the data/database.db entry of your device?

**Reply by xhemp** [reply]
> Can you provide the data/database.db entry of your device?

Sure!
I have removed it from the network but I will see if I can reinstall it, then readd it. It's almost like I updated it with the wrong firmware but my understanding is that wouldn't be allowed?
Is there anything else you need (logs, etc)?

**Reply by Koenkk** [reply]
How did you update it with the wrong firmware?

**Reply by xhemp** [reply]
That's the only thing I could think of seeing it's a 2 rocker switch being recognised as 1 rocker but I didn't do or try anyting special, I noticed one of the lights were not working and when I checked, I saw the device being recognised as a 1 rocker device. Been using this environment for 6 years with deConz, went to ZHA and now Z2M (which I am very happy with).

**Comment by xhemp** [comment]
Attached is the output from database.db for this switch:

`
  "id": 51,
  "type": "Router",
  "ieeeAddr": "0x00158d0002f06f44",
  "nwkAddr": 39764,
  "manufId": 4447,
  "manufName": "LUMI",
  "powerSource": "Mains (single phase)",
  "modelId": "lumi.ctrl_ln1.aq1",
  "epList": [
    1,
    2,
    3,
    5
  ],
  "endpoints": {
    "1": {
      "profId": 260,
      "epId": 1,
      "devId": 81,
      "inClusterList": [
        0,
        4,
        3,
        6,
        16,
        5,
        10,
        1,
        2
      ],
      "outClusterList": [
        25,
        10
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "65281": {
              "3": 29,
              "5": 69,
              "8": 4446,
              "9": 2048,
              "100": 0,
              "152": 0,
              "154": 0
            },
            "65521": 0,
            "modelId": "lumi.ctrl_ln1.aq1",
            "appVersion": 1,
            "manufacturerName": "LUMI",
            "powerSource": 1,
            "zclVersion": 1,
            "stackVersion": 2,
            "hwVersion": 18,
            "dateCode": "12-23-2021"
          }
        },
        "genOnOff": {
          "attributes": {
            "61440": 60511232,
            "onOff": 0
          }
        }
      },
      "binds": [],
      "configuredReportings": [],
      "meta": {}
    },
    "2": {
      "profId": 260,
      "epId": 2,
      "devId": 9,
      "inClusterList": [
        12
      ],
      "outClusterList": [
        12,
        4
      ],
      "clusters": {
        "genAnalogInput": {
          "attributes": {
            "presentValue": 0
          }
        }
      },
      "binds": [],
      "configuredReportings": [],
      "meta": {}
    },
    "3": {
      "epId": 3,
      "inClusterList": [],
      "outClusterList": [],
      "clusters": {},
      "binds": [],
      "configuredReportings": [],
      "meta": {}
    },
    "5": {
      "epId": 5,
      "inClusterList": [],
      "outClusterList": [],
      "clusters": {
        "genMultistateInput": {
          "attributes": {
            "presentValue": 1
          }
        }
      },
      "binds": [],
      "configuredReportings": [],
      "meta": {}
    }
  },
  "appVersion": 1,
  "stackVersion": 2,
  "hwVersion": 18,
  "dateCode": "12-23-2021",
  "zclVersion": 1,
  "interviewCompleted": true,
  "meta": {},
  "lastSeen": 1651590014024,
  "defaultSendRequestWhen": "immediate"
}`
For the sake of it, I also added it to an Aqara Hub (first one) and it also reports as a 1 rocker even though it's a 2 (QBKG12LM).
Weird.

**Reply by Koenkk** [reply]
The device definitely reports the wrong modelID, could you stop z2m, change the modelId in the database.db from `lumi.ctrl_ln1.aq1` to `lumi.ctrl_ln2.aq1` and start again, does it work now?

**Comment by xhemp** [comment]
Thanks, Koenkk. I tried but it didn't work, I have no idea what happened there but thanks for the help!
